src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ActivityLogRazorLogger.cs

```diff
@@ -49,6 +49,19 @@ public override void LogVerbose(string message)
             Log(__ACTIVITYLOG_ENTRYTYPE.ALE_INFORMATION, message);
         }
 
+        public async override void SetUIContext(Guid uiContextGuid, bool isActive)
+        {
+            await _joinableTaskFactory.SwitchToMainThreadAsync();
+
+            var monitorSelection = _serviceProvider.GetService(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
```

Comment:
Should this be cached? We don't cache the `SVsActivityLog` from the log method, but maybe that's just older code?

Reply:
Does this only happen once during project initialization?

Reply:
No, it would happen every time we serialize to `project.razor.json`.

Reply:
I updated the bit that publishes this to make it only publish once(ish). If we end up using UIContexts elsewhere we'll still want to do "the right thing" though.